### PR TITLE
style: enabled `InnerAssignment` in checkstyle

### PR DESCRIPTION
**Pull request by CodaBlurd**
Related to #5119

### CODE REVIEW

File: src/main/java/com/thealgorithms/dynamicprogramming/KnapsackMemoization.java [hunk 1]
```diff
@@ -40,8 +40,16 @@ int solveKnapsackRecursive(int capacity, int[] weights, int[] profits, int numOf
             dpTable[numOfItems][capacity] = solveKnapsackRecursive(capacity, weights, profits, numOfItems - 1, dpTable);
             return dpTable[numOfItems][capacity];
         } else {
-            // Return value of table after storing
-            return dpTable[numOfItems][capacity] = Math.max((profits[numOfItems - 1] + solveKnapsackRecursive(capacity - weights[numOfItems - 1], weights, profits, numOfItems - 1, dpTable)), solveKnapsackRecursive(capacity, weights, profits, numOfItems - 1, dpTable));
+            // case 1. include the item, if it is less than the capacity
+            int includeCurrentItem = profits[numOfItems - 1] + solveKnapsackRecursive(capacity - weights[numOfItems - 1], weights, profits, numOfItems - 1, dpTable);
+
+
```
Comment (vil02) [inline src/main/java/com/thealgorithms/dynamicprogramming/KnapsackMemoization.java:46]:
```suggestion
```

**Review by vil02 — CHANGES_REQUESTED** [review]

File: src/main/java/com/thealgorithms/dynamicprogramming/KnapsackMemoization.java [hunk 2]
```diff
@@ -40,8 +40,15 @@ int solveKnapsackRecursive(int capacity, int[] weights, int[] profits, int numOf
             dpTable[numOfItems][capacity] = solveKnapsackRecursive(capacity, weights, profits, numOfItems - 1, dpTable);
             return dpTable[numOfItems][capacity];
         } else {
-            // Return value of table after storing
-            return dpTable[numOfItems][capacity] = Math.max((profits[numOfItems - 1] + solveKnapsackRecursive(capacity - weights[numOfItems - 1], weights, profits, numOfItems - 1, dpTable)), solveKnapsackRecursive(capacity, weights, profits, numOfItems - 1, dpTable));
+            // case 1. include the item, if it is less than the capacity
+            int includeCurrentItem = profits[numOfItems - 1] + solveKnapsackRecursive(capacity - weights[numOfItems - 1], weights, profits, numOfItems - 1, dpTable);
```
Comment (vil02) [inline src/main/java/com/thealgorithms/dynamicprogramming/KnapsackMemoization.java:44]:
```suggestion
            final int includeCurrentItem = profits[numOfItems - 1] + solveKnapsackRecursive(capacity - weights[numOfItems - 1], weights, profits, numOfItems - 1, dpTable);
```

File: src/main/java/com/thealgorithms/dynamicprogramming/KnapsackMemoization.java [hunk 3]
```diff
@@ -40,8 +40,15 @@ int solveKnapsackRecursive(int capacity, int[] weights, int[] profits, int numOf
...
+
+            // case 2. exclude the item if it is more than the capacity
+            int excludeCurrentItem = solveKnapsackRecursive(capacity, weights, profits, numOfItems - 1, dpTable);
```
Comment (vil02) [inline src/main/java/com/thealgorithms/dynamicprogramming/KnapsackMemoization.java:47]:
```suggestion
            final int excludeCurrentItem = solveKnapsackRecursive(capacity, weights, profits, numOfItems - 1, dpTable);
```

**Review by vil02 — CHANGES_REQUESTED** [review]

File: src/main/java/com/thealgorithms/searches/UnionFind.java [hunk 4]
```diff
@@ -25,7 +25,10 @@ public int find(int i) {
             return i;
         }
 
-        return p[i] = find(parent);
+        int result = find(parent);
```
Comment (vil02) [inline src/main/java/com/thealgorithms/searches/UnionFind.java:28]:
```suggestion
        final int result = find(parent);
```

**Review by vil02 — APPROVED** [review]
Thanks!

I allowed myself to push few commits into your branch, I hope you don't mind.

**Comment by CodaBlurd** [comment]
No not at all. Thank you for helping me learn and improve!

**Comment by vil02** [comment]
Feel free to create PRs witch add missing tests, like #5160. This is the best way to find and fix bugs.

**Pull request merged**